### PR TITLE
Added known issue about pytest issues

### PR DESCRIPTION
**Pull request by astrofrog**
This tries to summarize the various pytest issues we've been having.

### CODE REVIEW

File: docs/known_issues.rst [hunk 1]
```diff
@@ -347,3 +347,24 @@ processing and you just want docstrings to work, this may be
...
+
+However, there is also an unrelated issue with pytest 4.0 and later which can
+also cause issues when collecting tests - in this case, the symptom is that the
+test collection hangs and/or appears to run the tests recursively. If you are
+maintaining a package that was created using the astropy
+`package template <http://github.com/astropy/package-template>`_, then
+this can be fixed by updating to the latest version of the ``_astropy_init.py``
+file. The root cause of this issue is that pytest now tries to pick up the
+top-level ``test()`` function as a test, so we need to make sure that we set a
+``test.__test__`` attribute on the function to ``False``.
```
Comment (bsipocz) [inline docs/known_issues.rst]: Could you add the third issue as well namely than in python 4.0+ the deprecation warning for `pytest_plugins` has turned into an error, so the LTS branch can never be pytest 4.0 compatible. Branch 3.0.x might be also affected, I'm not sure.

I think adding `pytest-astropy` and thus the astropy plugins as separate packages would fix this in addition of switching to implicit imports from `astropy.tests.pytest_plugins` (the latter may not be needed, I'm not sure any more given that I was messing with a combination of all of the issues above)


https://github.com/astropy/astropy/issues/7782

File: docs/known_issues.rst [hunk 2]
```diff
@@ -347,3 +347,24 @@ processing and you just want docstrings to work, this may be
...
+
+Compatibility issues with pytest 3.7 and later
+----------------------------------------------
+
+Due to a bug in `pytest <http://www.pytest.org>`_ related to test collection,
+the tests for the core astropy package for version 2.0.x (LTS), and for packages
+using the core package's test infrastructure and being tested against 2.0.x
+(LTS) will not be executed correctly with pytest 3.7, 3.8, or 3.9, and therefore
+pytest 3.6 or earlier, or pytest 4.0 or later should be used. The symptom of
+this bug is that no tests or only tests in rst files are collected. Note that
```
Comment (pllim) [inline docs/known_issues.rst]: Nitpick:
```suggestion
this bug is that no tests or only tests in RST files are collected. Note that
```

**Comment by astrofrog** [comment]
@bsipocz - i think I've addressed your comment - let me know what you think.

File: docs/known_issues.rst [hunk 3]
```diff
@@ -347,3 +347,28 @@ processing and you just want docstrings to work, this may be
...
+Compatibility issues with pytest 3.7 and later
+----------------------------------------------
+
+Due to a bug in `pytest <http://www.pytest.org>`_ related to test collection,
+the tests for the core astropy package for version 2.0.x (LTS), and for packages
+using the core package's test infrastructure and being tested against 2.0.x
+(LTS) will not be executed correctly with pytest 3.7, 3.8, or 3.9. The symptom
+of this bug is that no tests or only tests in RST files are collected. In
+addition, astropy 2.0.x (LTS) is not compatible with pytest 4.0 and above
+as in this case deprecation warnings from pytest can cause tests to fail.
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply docs/known_issues.rst]: Fixed

**Comment by bsipocz** [comment]
Merging this now, any more adjustments can be done in a follow-up.

**Comment by bsipocz** [comment]
Thank you @astrofrog, as @pllim said above this will save us tons of time.

**Pull request merged**